sdk/src/trace/span.cc

```diff
@@ -58,18 +58,15 @@ trace_api::SpanId GenerateRandomSpanId()
 }
 
 Span::Span(std::shared_ptr<Tracer> &&tracer,
-           std::shared_ptr<SpanProcessor> processor,
```

Comment:
There is a reason why the processor was put onto the span: this is to guarantee that the span/recordable gets ended on the same processor it was created with.

Otherwise, this could happen:
1. A span/recordable _a_ is created from processor/exporter A.
2. The processor/exporter is changed, maybe replaced by a composite processor/exporter B.
3. When the span/recordable _a_ is ended it is passed to the processor/exporter B and cannot be handled in this context (as it was created from processor/exporter A).

We should decide if that's still a use case we want to support. Just putting it out there that this was considered in the initial design.

Reply:
Ah, I had a different proposal where a processor asks for its recordable back later and receives a nullptr (or other "false" flag).

I can revert this for the other design.  I think this *is* a use case that needs support, there's just more than one option for how to do it.

Reply:
I don't mind which option we take. I'm in favor though of guaranteeing that the `Recordable` passed to the exporter is of the correct type.